### ISSUE TITLE
An option to hide SponsorBlock button

### ISSUE THREAD

**Issue by ARBoyGo** [libre-tube/LibreTube#2206]
### Describe your suggested feature

An option to Toggle on/off in settings to hide SponsorBlock button in videoplayer.

I like clean look and that button looks so bad, Also it doesn't look like I need that option. 

There is already an issue without proper format #2201

### Acknowledgements

- [ ] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by ARBoyGo** [comment]
As implemented with #2117 , a button has been added to the player to temporarily toggle sponsorblock, which is really a helpful feature.

Making the same an option to enable or disable would be better providing users the control if they want that button at the player or not
If someone don't want that button, he/she can turn it off from the settings.

-  @VedxP

**Comment by akano12** [comment]
I don't really understand why do people keep complaining about the smallest tiniest unnecessary details. 
The button you're complaining about is only visible for few seconds and only when you touch the player (and by touching the player - in case you didn't notice - makes other buttons visible, not just the hide sponsored parts).

**Comment by akano12** [comment]
This is going to be long but hopefully you'll get what I'm saying.

First of all -since most of the users who are complaining on every single small detail are not developers- I'm a developer and i can assure you that @Bnyro has been really generous and kind to most of you by adding whatever features and changes you're asking for. He's also been wayyyyy too kind by introducing new features **while keeping and maintaining the changes instead of completely getting rid of old code** (Introducing new layouts/changes while adding an option in settings to revert the changes). If it was for another developer (like me for example), I'll gladly close the issue and tell you to fork the repo.

Some of you will think I'm overreacting but it's the opposite, go and check the issues and you'll see that most of the issues here are "I don't like this" and "I don't like that" and some people are even asking for features to be removed because they don't like them (#2142). If you don't like _feature x_ for example, there are hundreds and even thousands using it without complaining.

Bottom of the line is, the world doesn't revolve around you, the code is open source and you're free to clone, modify, customize and build the project to suit your **PERSONAL PREFERENCES**. If you think programming/coding is hard to learn then hopefully you understood why you shouldn't be complaining in the first place, because programming is the easiest part, **maintaining the project while having a lot of options for users to customize the app is the hardest** because most of the things you're complaining about and asking to change aren't possible, there will be a hell lot of conflicts and lot of time being spent trying to figure out some workarounds. 

Take this project as an example ([XprivacyLua](https://github.com/M66B/XPrivacyLua)) (from the same dev of FairEmail and NetGuard), according to the estimates of the dev the number of active users have dropped to ~500-1000 users and he still supported the app by fixing bugs and adding compatibility after each android release, yet people were still ungrateful, complaining and hating till he finally gave up the development of the app.
Be thankful with what you have.

**Comment by Bnyro** [comment]
Thanks for these honest and kind words @akano12!
Unfortunately a lot of people forget that it's an open source project and that it might be to the developer's liking if users are creating tons of non-sense issues as well as duplicates while ticking they searched other issues and it's **NOT** a duplicate. A good example for that is #1919, there were more than ten duplicated of that issue and it was hard to stay kind and refer them to the already closed issue. All developers contributing here are doing so in their free time without getting paid or any other benefit for their invested time and efforts. At least one could expect the users to take some time when opening issues as well as including details, drafts or arguments why xy seems like a good idea to them, but unfortunately a bunch of people create issues with a simple screenshot or just a heading and expect developers to take tons of minutes or hours of their time just to do the community/person a favor and add their suggested feature. I won't accuse anyone in specific here, but the general situation should be clear.
That doesn't only apply for LibreTube, the same thing happens at Piped regularly with users complaining that nobody would care about them or looked at their issues and end up leaving the project and using an other alternative instead, what they usually state out in a long message out of hate and intolerance (though I doubt other projects would treat them better than we do). That's why @FireMasterK and I decided to add a Code of Conduct to the repos in the future and ignore, close or lock anything that violates them. We're not a company that has to fit the users needs, we're a FOSS project.
However, I think that we should have started doing so already way earlier which could potentially have saved a lot of time worrying about it that could have been used for developing or doing other IRL stuff instead (which is still part of every dev although some people might not believe it). Until we have a CoC it might be a good idea to just open a new issue and pin it to the repo to potentially improve the situation or at least have warned people that fit into the previously described category.
Just to make sure nobody gets me wrong, I do appreciate any constructive and forward-bringing interaction with users/the community that are well over-thought and aware of what FOSS actually means for the project, but that's unfortunately not the case for everybody I've came across so far. Anything I wrote here is not related to this issue or its participants and should be rather seen as "summary" of the overall situation.
There are communities that work way better than ours currently (e.g. some Linux distros like Garuda Linux), I wish we could be like that too one day.

**Comment by xIUPITERx** [comment]
I agree with you, of course, but how is one supposed to know that so much "small stuff" is undesirable if one has no previous experience with GitHub etc.? It's understandable that developers are annoyed by people like me, but without a guideline of what's wanted, I can't predict such things.
Besides, others and I have already asked whether there are too many requests, which (in my opinion) was not answered in the affirmative.
So if you @akano12 have a problem with the current handling on GitHub then feel free to discuss it with Byrno directly, but without attacking us for something that is apparently not obvious to many (users).
If this GitHub community wasn't so active, I would never have invested time in it myself. (Yes, it was only a theme, but that's all I know how to do).

**Comment by akano12** [comment]
Now before i reply to your points, let me show you how this small brilliant suggestion you had up in this thread would be translated into an implementation:

>The Icon should only be displayed if the current video contains any skippable segments.

Now what if i had custom selected segments to be skipped? Devs would have to check if SponsorBlock have an api where they could send my custom preferences in an api request then receive a response from the server just to decide whether to show or hide the sponsor block button. And all of this must be done the moment you click on the video and before you the video starts playing. 

Also what if i whitelisted a sponsor category? In addition to the whole unnecessary check there will be another entirely unnecessary check to see if the segments present in the video are whitelisted or not, and again this should be done before the video start playing. 

also let's not forget the scenarios where the server might be down, or region blocked or temporary unavailable, your video won't play because LT didn't receive a response whether to hide or show the sponsorblock button

**Comment by akano12** [comment]
Now moving on to the remaining point: 
1- Nobody said you're annoying.
2- Designing a theme or fixing a type, it's still a contribution and it's more than welcomed.
3- "Small Stuff" are always welcomed to be suggested, but nagging about "Small Stuff" is not recommended.
4- I'm pretty sure if you were the maintainer of the project and you kept seeing people nagging about everything here and there, you'll tell them to shut up and fork their own repo.
**5- As I mentioned earlier, Bnyro has been kind enough to keep some features as an option to select ever after being replaced with newer implementation. And this is the type of response you'd get from a lot of other devs/contributors when you keep nagging about something**
![2022-12-04_01-32](https://user-images.githubusercontent.com/50032603/205467038-1a64e63f-19b9-42b1-87df-4674c13c989d.png)

**Comment by ARBoyGo** [comment]
Honestly, I didn't need this, But I was discussing about this(#2117) with bnyro in matrix, That's the reason I wrote this feature request. Also I personally doesn't want to turn off sponsor block, Forgot it. 

Now, I feel like why did I wrote this 🤦. 

I'm so lazy to write a issue, also I know as a developer it's hard 😭 to see so many feature request. 


LT is a perfect competitor to YT with all features it has. 
Let's forget this feature request... If anyone want this feature please do PR.

**Comment by ARBoyGo** [comment]
I Think Icon Redesign is better.... Let's forget this...

**Comment by Bnyro** [comment]
> Some thoughts to add:
> ![Screenshot_20221202_102127_LibreTube Debug.png](https://user-images.githubusercontent.com/97352673/205266165-d151e8b7-7afa-4f35-9fad-15021ee84048.png)
> 1. I find the new button extremely useful, but could it be made smaller
> 2. Use a more minimalistic shield icon, the detached border around it doesn't quite fit in with the app's design language
> 3. The Icon should only be displayed if the current video contains any skippable segments.
> 
> Whether the button should switch between on and off or between auto skip and manual is probably a matter of opinion.
>
Well, I don't think that these suggestions in particular are too bad and somewhat agree on them.
However, I won't invest any time in searching for shield icons in the web for example, so what I would wish here for example is that the comment already contained some SVG icon suggestions so that it would be easy for whoever works on it to just use these. Apart from that the 3rd point is a bug as it should have already been working like that in theory (unfortunately it doesn't).

Disabling the button in general doesn't make sense to me: If you don't want to have it disable SponsorBlock in general (not related to the referenced comment though)!

And just to give another highlight of useful comments: https://github.com/libre-tube/LibreTube/issues/2218#issuecomment-1336304278.
